Server/src/main/java/org/xdi/oxauth/service/RedirectionUriService.java

```diff
@@ -74,7 +74,7 @@ public String validateRedirectionUri(String clientIdentifier, String redirection
 
                 if (StringUtils.isNotBlank(redirectionUri)) {
                     log.debug("Validating redirection URI: clientIdentifier = {}, redirectionUri = {}, found = {}",
-                            clientIdentifier, redirectionUri, redirectUris.length);
+                            clientIdentifier, redirectionUri, redirectUris != null ? redirectUris.length : 0);
 
                     final String redirectUriWithoutParams = uriWithoutParams(redirectionUri);
 
```

Comment:
Here in line 81 we have `redirectUris` in for loop. If it is null then in line 81 we will get NPE. It means we have to put null check somewhere above and handle it correctly.

Reply:
You're right. I've put the `null` guard in the enclosing `if`, have a look in d217dd80cdd5f8ce597bfc2e383451ca4b1a9b81